### PR TITLE
Update calls to createDV + updateDV with AssociatedModels

### PR DESCRIPTION
**Pull request by srrokib**
### Notes
- Update calls to `create_detector_version` and `update_detector_version` to accomodate AssociatedModels
- Add unit tests for parameter verification 
 - create + update DV called with 1 external model and 1 model version
 - create + update DV called with no AssociatedModels

#### Unit Tests

`./run_unit_tests`

```
...
...
Required test coverage of 89.0% reached. Total coverage: 89.64%

========================================================================================================= 215 passed in 31.64s =========================================================================================================
```

#### Contract Tests

`./cfn_server detector` and `cfn test`

```
...
...

============================================================================================== 14 passed, 2 skipped in 151.42s (0:02:31) ===============================================================================================
```

### Pre-commit

`pre-commit run --all-files `

```
black....................................................................Passed
flake8...................................................................Passed
Check for case conflicts.................................................Passed
Detect Private Key.......................................................Passed
Fix End of Files.........................................................Passed
Mixed line ending........................................................Passed
Trim Trailing Whitespace.................................................Passed
Pretty format JSON.......................................................Passed
Check for merge conflicts................................................Passed
Check Yaml...............................................................Passed
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-frauddetector-detector/src/aws_frauddetector_detector/tests/helpers/test_update_worker_helpers.py [hunk 1]
```diff
@@ -407,6 +409,16 @@ def test_update_detector_version_for_detector_update(monkeypatch):
     # Assert
     assert mock_call_describe_detector.call_count == 2
     assert len(detector_versions_to_delete) == 1  # we create a new DV when existing DV is not DRAFT
+    mock_call_create_detector_version.assert_called_once_with(
+        frauddetector_client=mock_afd_client,
+        detector_id=fake_model.DetectorId,
+        rules=ANY,
```
Comment (privetjr) [inline aws-frauddetector-detector/src/aws_frauddetector_detector/tests/helpers/test_update_worker_helpers.py:415]: I was wondering if there was a version of `Any` in unittest, nice!

File: aws-frauddetector-detector/src/aws_frauddetector_detector/tests/helpers/test_update_worker_helpers.py [hunk 2]
```diff
@@ -537,7 +662,7 @@ def test_validate_dependencies_for_inline_event_type_update(monkeypatch):
     assert mock_put_inline_entity_type.call_count == 1
     assert mock_create_inline_event_variable.call_count == 0
     assert mock_call_update_variable.call_count == 2
-    assert mock_update_tags.call_count == 7
+    assert mock_update_tags.call_count == 8
```
Comment (privetjr) [inline aws-frauddetector-detector/src/aws_frauddetector_detector/tests/helpers/test_update_worker_helpers.py:665]: Why does this change??

File: aws-frauddetector-detector/src/aws_frauddetector_detector/tests/helpers/test_update_worker_helpers.py [hunk 3]
```diff
@@ -595,7 +720,7 @@ def test_validate_dependencies_for_inline_event_type_update_referenced_dependenc
     assert mock_put_inline_entity_type.call_count == 1  # we call put entity type regardless for simplicity
     assert mock_create_inline_event_variable.call_count == 0
     assert mock_call_update_variable.call_count == 2  # we call update variable regardless for simplicity
-    assert mock_update_tags.call_count == 7
+    assert mock_update_tags.call_count == 8
```
Comment (privetjr) [inline aws-frauddetector-detector/src/aws_frauddetector_detector/tests/helpers/test_update_worker_helpers.py:723]: Why does this change??

**Review by privetjr — COMMENTED** [review]
This looks good to me, but I'm worried that somehow the number of calls to `update_tags` has increased... Any ideas why?

**Review by privetjr — APPROVED** [review]
As per offline discussion, the surprise w.r.t. update_tags.call_count is not breaking. Approving, and investigating separately.

**Review by sahilper — APPROVED** [review]

**Pull request merged**